examples/hosts/iframe-host/src/inframehost.ts

```diff
@@ -90,29 +38,19 @@ export class IFrameOuterHost {
         // set as non-user client
         this.loader = new Loader({
             ...hostConfig,
-            codeLoader: new ProxyCodeLoader(),
         });
     }
 
-    public async load(request: IRequest, iframe: HTMLIFrameElement): Promise<Container> {
-        const proxy = await IFrameDocumentServiceProxyFactory.create(
+    public async loadOuterProxy(iframe: HTMLIFrameElement): Promise<void> {
+        const outerProxy =  new DocumentServiceFactoryProxy(
             MultiDocumentServiceFactory.create(this.hostConfig.documentServiceFactory),
-            iframe,
             this.hostConfig.options,
-            MultiUrlResolver.create(this.hostConfig.urlResolver));
-
-        await proxy.createDocumentServiceFromRequest(request);
-
-        // don't try to connect until the iframe does, so they get existing false
-
-        await new Promise((resolve) => setTimeout(() => resolve(), 200));
-
-        const container = await this.loader.resolve(request);
-        if (!container.getQuorum().has("code")) {
-            // we'll never propose the code, so wait for them to do it
-            await new Promise((resolve) => container.once("contextChanged", () => resolve()));
-        }
+            iframe,
+        );
+        void outerProxy;
+    }
 
-        return container;
+    public async getContainerForRequest(request: IRequest): Promise<Container> {
```

Comment:
this shouldn't exist here, as the iframehost should never load or be exposed to the inner code. i would have expected a createDetached here that returned some kind of proxy to a detached container inside the iframe on which the outer could call attach

Reply:
to start with, it's probably easier to just have the inner just create detached, and call attach, which get's proxied to the outer. what i call out above could come i follow up

---
In reply to: [527049230](https://github.com/microsoft/FluidFramework/pull/4404#discussion_r527049230) [](ancestors = 527049230)

Reply:
I created a container proxy layer and moved the attach call to the outer host.  Wasn't too clear on what you meant in the second suggestion though, because calling an `attach` that's been proxied to the outer host would mean the real container exists in outer, which didn't seem like what we're going for